File: python/leetcode/118.py
class Solution:
    def generate(self, numRows):
        res = [[0 for j in range(i+1)] for i in range(numRows)]
        for i in range(numRows):
            for j in range(i+1):
                if j == 0 or j == i:
                    res[i][j] = 1
                else:
                    res[i][j] = res[i-1][j]+res[i-1][j-1]
        return res

File: cpp/leetcode/118.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        if(n < 1) return {};
        vector<vector<int>> res(n, vector<int>());
        for(int i = 0; i < n; i++){
            for(int j = 0; j <= i; j++){
                if(j == 0 || j == i) res[i].push_back(1);
                else res[i].push_back(res[i-1][j-1] + res[i-1][j]);
            }
        }
        return res;
    }
};